From cb53ac3d5effc5bdd36ad0b23ec83bd8a09d3797 Mon Sep 17 00:00:00 2001
From: Denis Baranov <baraka@etersoft.ru>
Date: Fri, 4 Jan 2013 19:05:06 +0400
Subject: [PATCH 5/6] Build new version opennx-0.16-eter18.svn724

---
 .gear/opennx.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/.gear/opennx.spec b/.gear/opennx.spec
index f89c79f..bc8aff1 100644
--- a/.gear/opennx.spec
+++ b/.gear/opennx.spec
@@ -4,7 +4,7 @@
 Summary: An OpenSource NX client
 Name: opennx
 Version: 0.16
-Release: eter17.svn724
+Release: eter18.svn724
 License: LGPL/GPL
 Group: Networking/Remote access
 Url: http://sourceforge.net/projects/opennx
@@ -123,6 +123,9 @@ install -m 644 etc/*.rules %buildroot%_sysconfdir/udev/rules.d
 %endif
 
 %changelog
+* Fri Jan 04 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter18.svn724
+- Add wizard button to start menu
+
 * Thu Jan 03 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter17.svn724
 - update from trunk opennx-0.16.0.724
 
-- 
1.8.1.1

