From ffab0b7adbb776aa9bdbe87bd40c6656159362b4 Mon Sep 17 00:00:00 2001
From: Denis Baranov <baraka@etersoft.ru>
Date: Thu, 17 Jan 2013 19:49:45 +0400
Subject: [PATCH 6/7] Add copy icon in spec

---
 .gear/opennx.spec | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/.gear/opennx.spec b/.gear/opennx.spec
index bc8aff1..7612f31 100644
--- a/.gear/opennx.spec
+++ b/.gear/opennx.spec
@@ -86,15 +86,15 @@ subst "s|/usr/NX/bin|%_bindir|g" %buildroot%_desktopdir/*
 
 install -d %buildroot{%_niconsdir,%_miconsdir,%_liconsdir}
 for f in nx opennx-admin opennx-wizard ; do
-    convert -size 16x16 ./extres/scalable/apps/$f.svg %buildroot%_miconsdir/$f.png
-    convert -size 32x32 ./extres/scalable/apps/$f.svg %buildroot%_niconsdir/$f.png
-    convert -size 48x48 ./extres/scalable/apps/$f.svg %buildroot%_liconsdir/$f.png
+    install -m 644 ./extres/16x16/apps/$f.png %buildroot%_miconsdir/$f.png
+    install -m 644 ./extres/32x32/apps/$f.png %buildroot%_niconsdir/$f.png
+    install -m 644 ./extres/48x48/apps/$f.png %buildroot%_liconsdir/$f.png
 done
 
 install -d %buildroot%_iconsdir/hicolor/{16x16,32x32,48x48}/mimetypes/
-convert -background none -size 16x16 ./extres/scalable/mimetypes/nx-desktop.svg %buildroot%_iconsdir/hicolor/16x16/mimetypes/nx-desktop.png
-convert -background none -size 32x32 ./extres/scalable/mimetypes/nx-desktop.svg %buildroot%_iconsdir/hicolor/32x32/mimetypes/nx-desktop.png
-convert -background none -size 48x48 ./extres/scalable/mimetypes/nx-desktop.svg %buildroot%_iconsdir/hicolor/48x48/mimetypes/nx-desktop.png
+install -m 644 ./extres/16x16/mimetypes/nx-desktop.png %buildroot%_iconsdir/hicolor/16x16/mimetypes/nx-desktop.png
+install -m 644 ./extres/32x32/mimetypes/nx-desktop.png %buildroot%_iconsdir/hicolor/32x32/mimetypes/nx-desktop.png
+install -m 644 ./extres/48x48/mimetypes/nx-desktop.png %buildroot%_iconsdir/hicolor/48x48/mimetypes/nx-desktop.png
 
 %if %with_usbip
 install -d -m 755 %buildroot%_sysconfdir/udev/rules.d
-- 
1.8.1.1

