From d7a6c24e9cc7704b9a815f7bcde094c60e32f780 Mon Sep 17 00:00:00 2001
From: Denis Baranov <baraka@etersoft.ru>
Date: Thu, 17 Jan 2013 19:51:12 +0400
Subject: [PATCH 07/18] Build new version opennx-0.16-eter19.svn724

---
 .gear/opennx.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/.gear/opennx.spec b/.gear/opennx.spec
index 7612f31..f368023 100644
--- a/.gear/opennx.spec
+++ b/.gear/opennx.spec
@@ -4,7 +4,7 @@
 Summary: An OpenSource NX client
 Name: opennx
 Version: 0.16
-Release: eter18.svn724
+Release: eter19.svn724
 License: LGPL/GPL
 Group: Networking/Remote access
 Url: http://sourceforge.net/projects/opennx
@@ -123,6 +123,9 @@ install -m 644 etc/*.rules %buildroot%_sysconfdir/udev/rules.d
 %endif
 
 %changelog
+* Thu Jan 17 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter19.svn724
+- Add copy icon in spec
+
 * Fri Jan 04 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter18.svn724
 - Add wizard button to start menu
 
-- 
1.8.3.4

