From 795b22867e4a34dda0d64b7e98d0772358a01596 Mon Sep 17 00:00:00 2001
From: Denis Baranov <baraka@etersoft.ru>
Date: Fri, 18 Jan 2013 00:50:20 +0400
Subject: [PATCH 11/14] Build new version opennx-0.16-eter20.svn724

---
 .gear/opennx.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/.gear/opennx.spec b/.gear/opennx.spec
index f368023..2986f32 100644
--- a/.gear/opennx.spec
+++ b/.gear/opennx.spec
@@ -4,7 +4,7 @@
 Summary: An OpenSource NX client
 Name: opennx
 Version: 0.16
-Release: eter19.svn724
+Release: eter20.svn724
 License: LGPL/GPL
 Group: Networking/Remote access
 Url: http://sourceforge.net/projects/opennx
@@ -123,6 +123,11 @@ install -m 644 etc/*.rules %buildroot%_sysconfdir/udev/rules.d
 %endif
 
 %changelog
+* Fri Jan 18 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter20.svn724
+- Translate of Resume button correct
+- Update translations
+- Add translate for Wizard button
+
 * Thu Jan 17 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter19.svn724
 - Add copy icon in spec
 
-- 
1.8.1.6

