From 1fe6e18398c294dc9ad43fcd127570997ed4e7c7 Mon Sep 17 00:00:00 2001
From: Denis Baranov <baraka@etersoft.ru>
Date: Mon, 10 Jun 2013 15:53:38 +0400
Subject: [PATCH 16/37] Build new version opennx-0.16-eter22.svn724

---
 .gear/opennx.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/.gear/opennx.spec b/.gear/opennx.spec
index 0d7318b..e3baf6b 100644
--- a/.gear/opennx.spec
+++ b/.gear/opennx.spec
@@ -4,7 +4,7 @@
 Summary: An OpenSource NX client
 Name: opennx
 Version: 0.16
-Release: eter21.svn724
+Release: eter22.svn724
 License: LGPL/GPL
 Group: Networking/Remote access
 Url: http://sourceforge.net/projects/opennx
@@ -103,6 +103,9 @@ install -m 644 etc/*.rules %buildroot%_sysconfdir/udev/rules.d
 %endif
 
 %changelog
+* Mon Jun 10 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter22.svn724
+- Add patch for translate
+
 * Mon Feb 11 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter21.svn724
 - Now all paths write to config after start
 
-- 
2.1.4

