From c4ba217a60e9a644d4cce1a81a9f39d2303b5c82 Mon Sep 17 00:00:00 2001
From: Vitaly Lipatov <lav@etersoft.ru>
Date: Mon, 5 Aug 2013 13:54:20 +0400
Subject: [PATCH 19/34] new build 0.16-eter23 (with rpmlog script)

---
 .gear/opennx.spec | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/.gear/opennx.spec b/.gear/opennx.spec
index e3baf6b..35613df 100644
--- a/.gear/opennx.spec
+++ b/.gear/opennx.spec
@@ -1,13 +1,16 @@
 # Enable USBIP support
 %define with_usbip %{?_with_usbip: 1} %{?!_with_usbip: 0}
 
-Summary: An OpenSource NX client
 Name: opennx
 Version: 0.16
-Release: eter22.svn724
+Release: eter23.svn724
+
+Summary: An OpenSource NX client
+
 License: LGPL/GPL
 Group: Networking/Remote access
 Url: http://sourceforge.net/projects/opennx
+
 Packager: Denis Baranov <baraka@altlinux.org>
 
 Source: ftp://ftp.etersoft.ru/pub/Etersoft/RX@Etersoft/1.1/sources/tarball/%name-%version.tar
@@ -103,6 +106,9 @@ install -m 644 etc/*.rules %buildroot%_sysconfdir/udev/rules.d
 %endif
 
 %changelog
+* Mon Aug 05 2013 Vitaly Lipatov <lav@altlinux.ru> 0.16-eter23.svn724
+- add Num Lock state as a parameter to be send
+
 * Mon Jun 10 2013 Denis Baranov <baraka@altlinux.ru> 0.16-eter22.svn724
 - Add patch for translate
 
-- 
2.1.4

