diff -urw pulseaudio-1.0-dev-366-g7d918-orig/src/daemon/daemon.conf.in pulseaudio-1.0-dev-366-g7d918/src/daemon/daemon.conf.in
--- pulseaudio-1.0-dev-366-g7d918-orig/src/daemon/daemon.conf.in	2011-04-30 14:56:13.823035998 +0200
+++ pulseaudio-1.0-dev-366-g7d918/src/daemon/daemon.conf.in	2011-06-16 17:18:33.548348404 +0200
@@ -37,7 +37,7 @@
 ; realtime-scheduling = yes
 ; realtime-priority = 5
 
-; exit-idle-time = 20
+exit-idle-time = -1
 ; scache-idle-time = 20
 
 ; dl-search-path = (depends on architecture)
